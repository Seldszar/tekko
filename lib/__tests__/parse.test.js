const parse = require("../parse");

describe("parse", () => {
  const testCases = [
    "",
    "@",
    "FOO",
    ":test ",
    ":test FOO",
    ":test FOO     ",
    ":! FOO",
    ":@ FOO",
    ":nick FOO",
    ":nick!user FOO",
    ":nick!user@host FOO",
    ":nick@host FOO",
    ":se.rv.er FOO",
    ":test!me@test.ing PRIVMSG #Test :This is a test",
    "PRIVMSG #Test :This is a test",
    ":test PRIVMSG foo :A string   with spaces   ",
    ":test     PRIVMSG    foo     :bar",
    ":test FOO bar baz quux",
    "FOO bar baz quux",
    "FOO    bar      baz   quux",
    "FOO bar baz quux :This is a test",
    ":test PRIVMSG #fo:oo :This is a test",
    "@test=super;single :test!me@test.ing FOO bar baz quux :This is a test",
    "@test=super;single     :test!me@test.ing    FOO   bar    baz   quux   :This is a test",
    "@test=super;single :  This is a test",
    "@test=super;single :",
    "@test=;single :",
    "@test=super\\shyper;single :",
    "@test=super\\r\\n\\\\\\:\\shyper\\;single :",
  ];

  for (const testCase of testCases) {
    test(`should parse "${testCase}"`, () => {
      let result;

      try {
        result = parse(testCase);
      } catch (error) {
        result = error;
      }

      expect(result).toMatchSnapshot();
    });
  }
});
