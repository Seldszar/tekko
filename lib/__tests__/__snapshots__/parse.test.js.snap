// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse should parse "" 1`] = `[ParseError: Invalid message]`;

exports[`parse should parse ":! FOO" 1`] = `
Object {
  "command": "FOO",
  "params": Array [],
  "prefix": null,
  "tags": Object {},
  "trailing": null,
}
`;

exports[`parse should parse ":@ FOO" 1`] = `
Object {
  "command": "FOO",
  "params": Array [],
  "prefix": null,
  "tags": Object {},
  "trailing": null,
}
`;

exports[`parse should parse ":nick FOO" 1`] = `
Object {
  "command": "FOO",
  "params": Array [],
  "prefix": Object {
    "host": null,
    "name": "nick",
    "user": null,
  },
  "tags": Object {},
  "trailing": null,
}
`;

exports[`parse should parse ":nick!user FOO" 1`] = `
Object {
  "command": "FOO",
  "params": Array [],
  "prefix": Object {
    "host": null,
    "name": "nick",
    "user": "user",
  },
  "tags": Object {},
  "trailing": null,
}
`;

exports[`parse should parse ":nick!user@host FOO" 1`] = `
Object {
  "command": "FOO",
  "params": Array [],
  "prefix": Object {
    "host": "host",
    "name": "nick",
    "user": "user",
  },
  "tags": Object {},
  "trailing": null,
}
`;

exports[`parse should parse ":nick@host FOO" 1`] = `
Object {
  "command": "FOO",
  "params": Array [],
  "prefix": Object {
    "host": "host",
    "name": "nick",
    "user": null,
  },
  "tags": Object {},
  "trailing": null,
}
`;

exports[`parse should parse ":se.rv.er FOO" 1`] = `
Object {
  "command": "FOO",
  "params": Array [],
  "prefix": Object {
    "host": null,
    "name": "se.rv.er",
    "user": null,
  },
  "tags": Object {},
  "trailing": null,
}
`;

exports[`parse should parse ":test     PRIVMSG    foo     :bar" 1`] = `
Object {
  "command": "PRIVMSG",
  "params": Array [
    "foo",
  ],
  "prefix": Object {
    "host": null,
    "name": "test",
    "user": null,
  },
  "tags": Object {},
  "trailing": "bar",
}
`;

exports[`parse should parse ":test " 1`] = `[ParseError: Invalid message]`;

exports[`parse should parse ":test FOO     " 1`] = `
Object {
  "command": "FOO",
  "params": Array [],
  "prefix": Object {
    "host": null,
    "name": "test",
    "user": null,
  },
  "tags": Object {},
  "trailing": null,
}
`;

exports[`parse should parse ":test FOO bar baz quux" 1`] = `
Object {
  "command": "FOO",
  "params": Array [
    "bar",
    "baz",
    "quux",
  ],
  "prefix": Object {
    "host": null,
    "name": "test",
    "user": null,
  },
  "tags": Object {},
  "trailing": null,
}
`;

exports[`parse should parse ":test FOO" 1`] = `
Object {
  "command": "FOO",
  "params": Array [],
  "prefix": Object {
    "host": null,
    "name": "test",
    "user": null,
  },
  "tags": Object {},
  "trailing": null,
}
`;

exports[`parse should parse ":test PRIVMSG #fo:oo :This is a test" 1`] = `
Object {
  "command": "PRIVMSG",
  "params": Array [
    "#fo:oo",
  ],
  "prefix": Object {
    "host": null,
    "name": "test",
    "user": null,
  },
  "tags": Object {},
  "trailing": "This is a test",
}
`;

exports[`parse should parse ":test PRIVMSG foo :A string   with spaces   " 1`] = `
Object {
  "command": "PRIVMSG",
  "params": Array [
    "foo",
  ],
  "prefix": Object {
    "host": null,
    "name": "test",
    "user": null,
  },
  "tags": Object {},
  "trailing": "A string   with spaces   ",
}
`;

exports[`parse should parse ":test!me@test.ing PRIVMSG #Test :This is a test" 1`] = `
Object {
  "command": "PRIVMSG",
  "params": Array [
    "#Test",
  ],
  "prefix": Object {
    "host": "test.ing",
    "name": "test",
    "user": "me",
  },
  "tags": Object {},
  "trailing": "This is a test",
}
`;

exports[`parse should parse "@" 1`] = `[ParseError: Invalid message]`;

exports[`parse should parse "@test=;single :" 1`] = `[ParseError: Invalid message]`;

exports[`parse should parse "@test=super;single     :test!me@test.ing    FOO   bar    baz   quux   :This is a test" 1`] = `
Object {
  "command": "FOO",
  "params": Array [
    "bar",
    "baz",
    "quux",
  ],
  "prefix": Object {
    "host": "test.ing",
    "name": "test",
    "user": "me",
  },
  "tags": Object {
    "single": true,
    "test": "super",
  },
  "trailing": "This is a test",
}
`;

exports[`parse should parse "@test=super;single :  This is a test" 1`] = `
Object {
  "command": "This",
  "params": Array [
    "is",
    "a",
    "test",
  ],
  "prefix": null,
  "tags": Object {
    "single": true,
    "test": "super",
  },
  "trailing": null,
}
`;

exports[`parse should parse "@test=super;single :" 1`] = `[ParseError: Invalid message]`;

exports[`parse should parse "@test=super;single :test!me@test.ing FOO bar baz quux :This is a test" 1`] = `
Object {
  "command": "FOO",
  "params": Array [
    "bar",
    "baz",
    "quux",
  ],
  "prefix": Object {
    "host": "test.ing",
    "name": "test",
    "user": "me",
  },
  "tags": Object {
    "single": true,
    "test": "super",
  },
  "trailing": "This is a test",
}
`;

exports[`parse should parse "@test=super\\r\\n\\\\\\:\\shyper\\;single :" 1`] = `[ParseError: Invalid message]`;

exports[`parse should parse "@test=super\\shyper;single :" 1`] = `[ParseError: Invalid message]`;

exports[`parse should parse "FOO    bar      baz   quux" 1`] = `
Object {
  "command": "FOO",
  "params": Array [
    "bar",
    "baz",
    "quux",
  ],
  "prefix": null,
  "tags": Object {},
  "trailing": null,
}
`;

exports[`parse should parse "FOO bar baz quux :This is a test" 1`] = `
Object {
  "command": "FOO",
  "params": Array [
    "bar",
    "baz",
    "quux",
  ],
  "prefix": null,
  "tags": Object {},
  "trailing": "This is a test",
}
`;

exports[`parse should parse "FOO bar baz quux" 1`] = `
Object {
  "command": "FOO",
  "params": Array [
    "bar",
    "baz",
    "quux",
  ],
  "prefix": null,
  "tags": Object {},
  "trailing": null,
}
`;

exports[`parse should parse "FOO" 1`] = `
Object {
  "command": "FOO",
  "params": Array [],
  "prefix": null,
  "tags": Object {},
  "trailing": null,
}
`;

exports[`parse should parse "PRIVMSG #Test :This is a test" 1`] = `
Object {
  "command": "PRIVMSG",
  "params": Array [
    "#Test",
  ],
  "prefix": null,
  "tags": Object {},
  "trailing": "This is a test",
}
`;
